cam2image/src/showimage.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+
+#include <opencv2/highgui/highgui.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_interfaces/msg/image.hpp>
+
+
+int
+encoding2mat_type(const std::string & encoding)
+{
+  if(encoding == "mono8") {
+    return CV_8UC1;
+  } else if(encoding == "rgb8") {
+    return CV_8UC3;
+  } else if(encoding == "mono16") {
+    return CV_16SC1;
+  } else if(encoding == "rgba8") {
+    return CV_8UC4;
+  } else {
+    throw std::runtime_error("Unsupported encoding type");
+  }
+}
```

Comment:
The same code already exists in the other cpp file. I would suggest moving it to a header and reusing it in both.